cmd/rook/ceph/osd.go

```diff
@@ -326,9 +329,22 @@ func removeOSDs(cmd *cobra.Command, args []string) error {
 		return errors.Wrapf(err, "failed to parse --preserve-pvc flag")
 	}
 
-	exitIfNotSafe := false
-	forceRemovalCallback := func(x int) (bool, bool) {
-		return forceOSDRemovalBool, exitIfNotSafe
+	promptIfNotSafeBool, err := strconv.ParseBool(promptIfNotSafe)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse --prompt-if-not-safe flag")
+	}
+
+	forceRemovalCallback := func(osdIDsToRemove int) (bool, bool) {
+		if promptIfNotSafeBool {
+			var answer string
+			logger.Infof("Are you sure you want to remove osd ID %d?", osdIDsToRemove)
+			fmt.Scanf("%s", &answer)
```

Comment:
How about moving the Scanf() call inside the method `PromptToContinueOrCancel()`